q2_composition/_ancombc.py

```diff
@@ -0,0 +1,146 @@
+              ' longer exist.')
+    for cmd in cmds:
+        if verbose:
+            print('\nCommand:', end=' ')
+            print(' '.join(cmd), end='\n\n')
+        subprocess.run(cmd, check=True)
+
+
+def ancombc(table: pd.DataFrame, metadata: qiime2.Metadata,
+            formula: str, p_adj_method: str = 'holm', prv_cut: float = 0.1,
+            lib_cut: int = 0, group: str = None, level_ordering: str = None,
+            struc_zero: bool = False, neg_lb: bool = False, tol: float = 1e-05,
+            max_iter: int = 100, conserve: bool = False,
+            alpha: float = 0.05) -> pd.DataFrame:
+
+    return _ancombc(
+        table=table,
+        metadata=metadata,
+        formula=formula,
+        p_adj_method=p_adj_method,
+        prv_cut=prv_cut,
+        lib_cut=lib_cut,
+        group=group,
+        level_ordering=level_ordering,
+        struc_zero=struc_zero,
+        neg_lb=neg_lb,
+        tol=tol,
+        max_iter=max_iter,
+        conserve=conserve,
+        alpha=alpha,
+    )
+
+
+def _column_validation(value, parameter, metadata):
+    if value not in metadata.columns:
+        raise ValueError('Value provided in the `%s` parameter was not found'
+                         ' in any of the metadata columns. Please make sure to'
+                         ' only include values that are present within the'
+                         ' metadata columns.'
+                         ' \n\n'
+                         ' Value that was not found as a metadata column: "%s"'
+                         % (parameter, value))
+
+
+def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
+             group, level_ordering, struc_zero, neg_lb, tol, max_iter,
+             conserve, alpha):
+
+    meta = metadata.to_dataframe()
+
+    # error on IDs found in table but not in metadata
+    missing_ids = []
+    for i in set(table.index):
+        if i not in set(meta.index):
+            missing_ids.append(i)
+
+    if not(set(table.index).issubset(set(meta.index))):
+        raise KeyError('Not all samples present within the table were found in'
+                       ' the associated metadata file. Please make sure that'
+                       ' all samples in the FeatureTable are also present in'
+                       ' the metadata.'
+                       ' Sample IDs not found in the metadata: %s'
+                       % missing_ids)
+
+    # column validation for the group parameter
+    if group is not None:
+        _column_validation(group, 'group', meta)
+    else:
+        group = ''
+
+    # column & level validation for the level_ordering parameter
+    if level_ordering is not None:
+        for i in level_ordering:
+            column = i.split('::')[0]
+            level_value = i.split('::')[1]
+
+            _column_validation(column, 'level_ordering', meta)
+
+            if level_value not in np.unique(meta[column].values):
```

Comment:
There ought to be a .unique() method on a pandas series as well here, which will do the same thing